homeassistant/components/overkiz/alarm_control_panel.py

```diff
@@ -0,0 +1,298 @@
+]
+
+SUPPORTED_DEVICES = {description.key: description for description in ALARM_DESCRIPTIONS}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Overkiz alarm control panel from a config entry."""
+    data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[OverkizAlarmControlPanel] = []
+
+    for device in data.platforms[Platform.ALARM_CONTROL_PANEL]:
+        if description := SUPPORTED_DEVICES.get(device.widget) or SUPPORTED_DEVICES.get(
+            device.ui_class
+        ):
+            entities.append(
+                OverkizAlarmControlPanel(
+                    device.device_url,
+                    data.coordinator,
+                    description,
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class OverkizAlarmControlPanel(OverkizDescriptiveEntity, AlarmControlPanelEntity):
+    """Representation of an Overkiz Alarm Control Panel."""
+
+    entity_description: OverkizAlarmDescription
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return self.entity_description.supported_features
+
+    @property
+    def state(self) -> str:
+        """Return the state of the device."""
+        return self.entity_description.fn_state(self.executor.select_state)
+
+    async def async_alarm_disarm(self, code: str | None = None) -> None:
+        """Send disarm command."""
+        await self.async_execute_command(
+            self.entity_description.alarm_disarm,
+            self.entity_description.alarm_disarm_args,
+        )
+
+    async def async_alarm_arm_home(self, code: str | None = None) -> None:
+        """Send arm home command."""
+        await self.async_execute_command(
+            self.entity_description.alarm_arm_home,
+            self.entity_description.alarm_arm_home_args,
+        )
+
+    async def async_alarm_arm_night(self, code: str | None = None) -> None:
+        """Send arm night command."""
+        await self.async_execute_command(
+            self.entity_description.alarm_arm_night,
+            self.entity_description.alarm_arm_night_args,
+        )
+
+    async def async_alarm_arm_away(self, code: str | None = None) -> None:
+        """Send arm away command."""
+        await self.async_execute_command(
+            self.entity_description.alarm_arm_away,
+            self.entity_description.alarm_arm_away_args,
+        )
+
+    async def async_alarm_trigger(self, code: str | None = None) -> None:
+        """Send alarm trigger command."""
+        await self.async_execute_command(
+            self.entity_description.alarm_trigger,
+            self.entity_description.alarm_trigger_args,
+        )
+
+    async def async_execute_command(self, command_name: str | None, *args: Any) -> None:
+        """Check command name and send to the executor."""
+        if command_name:
```

Comment:
We should raise an error instead of silently ignoring the fact that a service got called that wasn't supported. It's a bug because SUPPORTED_FEATURES should have prevented this service from being called?

Reply:
I wasn't sure about this one indeed. In theory this could never be called, however mypy would complain about the command_name being `None`, since not all functions are implemented by all alarm control panels.

Would you rather have that we raise an error if command name is None? (or should we trust the SUPPORTED_FEATURES)

Reply:
You can trust the supported features, we skip the entities otherwise: https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/service.py#L644

Reply:
You can use assert command_name for mypy.

Reply:
@epenet what would be cleaner? `assert command_name` or `cast(str, command_name)`? I added the assert for now.

Reply:
assert is correct.